packages/protocol/contracts/layer1/provers/GuardianProver.sol

```diff
@@ -144,58 +142,39 @@ contract GuardianProver is IVerifier, EssentialContract {
         emit GuardiansUpdated(version, _newGuardians);
     }
 
-    /// @dev Enables or disables proving auto pause.
-    /// @param _enable true to enable, false to disable.
+    /// @notice Enables or disables proving auto pause.
+    /// @param _enable True to enable, false to disable.
     function enableProvingAutoPause(bool _enable) external onlyOwner {
-        if (provingAutoPauseEnabled == _enable) revert GP_INVALID_STATUS();
+        require(provingAutoPauseEnabled != _enable, GP_INVALID_STATUS());
         provingAutoPauseEnabled = _enable;
 
         emit ProvingAutoPauseEnabled(_enable);
     }
 
     /// @notice Enables unlimited allowance for Taiko L1 contract.
-    /// param _enable true if unlimited allowance is approved, false to set the allowance to 0.
+    /// @param _enable True if unlimited allowance is approved, false to set the allowance to 0.
     function enableTaikoTokenAllowance(bool _enable) external onlyOwner {
         address tko = resolve(LibStrings.B_TAIKO_TOKEN, false);
         address taiko = resolve(LibStrings.B_TAIKO, false);
         IERC20(tko).approve(taiko, _enable ? type(uint256).max : 0);
     }
 
-    /// @dev Withdraws Taiko Token to a given address.
+    /// @notice Withdraws Taiko Token to a given address.
     /// @param _to The recipient address.
     /// @param _amount The amount of Taiko token to withdraw. Use 0 for all balance.
     function withdrawTaikoToken(address _to, uint256 _amount) external onlyOwner {
-        if (_to == address(0)) revert GV_ZERO_ADDRESS();
+        require(_to != address(0), GV_ZERO_ADDRESS());
 
         IERC20 tko = IERC20(resolve(LibStrings.B_TAIKO_TOKEN, false));
         uint256 amount = _amount == 0 ? tko.balanceOf(address(this)) : _amount;
         tko.transfer(_to, amount);
     }
 
-    /// @dev Called by guardians to approve a guardian proof
-    /// @param _meta The block's metadata.
+    /// @notice Called by guardians to approve a guardian proof (version 2).
+    /// @param _metaV2 The block's metadata (version 2).
     /// @param _tran The valid transition.
     /// @param _proof The tier proof.
-    /// @return approved_ True if the minimum number of approval is acquired, false otherwise.
-    function approve(
```

Comment:
`approveV2` is not yet implemented on the taiko-client side.
So maybe we need to keep this code or raise the priority of implementing `approveV2`.

Reply:
I think its supported? https://github.com/taikoxyz/taiko-mono/blob/main/packages/taiko-client/prover/proof_submitter/transaction/builder.go#L127

Reply:
Oh, I got this and https://github.com/taikoxyz/taiko-mono/issues/18162 mixed up...